codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/RubySymbolProvider.java

```diff
@@ -352,6 +351,7 @@ public Symbol operationShape(OperationShape shape) {
 
     @Override
     public Symbol resourceShape(ResourceShape shape) {
-        throw new CodegenException("Resources are not currently supported");
```

Comment:
I think we prefer to keep this error. Resource shapes would never belong in types.rb.

Reply:
> Nice! Thank you. What are the overall benefits of this?

I think it's simply adhering to the Smithy based code generators pattern. But in terms of performance/functionality, it still the same for now as it re-uses the existing code generators.

---

Keeping the error would cause code generation to fail as directed codegen tries to call `SymbolProvider.resourceShape(...)` behind the scene whenever it encounters a ResourceShape.

I have removed `definitionFile` for now until we have a better idea on ResourceShape handling. (Ruby codgen currently doesn't handle/write ResourceShapes)